### PR TITLE
Yet another apporach of copying logical timeline size during branch creation

### PR DESCRIPTION
**Pull request by knizhnik**
Alternative to #2117

### CODE REVIEW

**Comment by aome510** [comment]
Some tests fail:
```
2022-07-21T14:06:54.8984360Z failures:
2022-07-21T14:06:54.8984527Z     layered_repository::tests::test_traverse_ancestors
2022-07-21T14:06:54.8984739Z     layered_repository::tests::test_traverse_branches
2022-07-21T14:06:54.8984919Z     repository::tests::test_branch
2022-07-21T14:06:54.8985122Z     repository::tests::test_parent_keeps_data_forever_after_branching
2022-07-21T14:06:54.8985362Z     repository::tests::test_retain_data_in_parent_which_is_needed_for_child
2022-07-21T14:06:54.8985623Z     repository::tests::timeline_load_with_ancestor
```

**Comment by knizhnik** [comment]
> Some tests fail:
> 
> ```
> 2022-07-21T14:06:54.8984360Z failures:
> 2022-07-21T14:06:54.8984527Z     layered_repository::tests::test_traverse_ancestors
> 2022-07-21T14:06:54.8984739Z     layered_repository::tests::test_traverse_branches
> 2022-07-21T14:06:54.8984919Z     repository::tests::test_branch
> 2022-07-21T14:06:54.8985122Z     repository::tests::test_parent_keeps_data_forever_after_branching
> 2022-07-21T14:06:54.8985362Z     repository::tests::test_retain_data_in_parent_which_is_needed_for_child
> 2022-07-21T14:06:54.8985623Z     repository::tests::timeline_load_with_ancestor
> ```

Just fixed it.

File: pageserver/src/layered_repository.rs [hunk 1]
```diff
@@ -323,6 +324,20 @@ impl Repository for LayeredRepository {
                 ));
             }
         }
+        // Copy logical size from source timeline if we are branching on te last positon.
```
Comment (aome510) [inline pageserver/src/layered_repository.rs:327]:
```suggestion
        // Copy logical size from source timeline if we are branching on the last position.
```

File: pageserver/src/layered_repository.rs [hunk 2]
```diff
@@ -323,6 +324,20 @@ impl Repository for LayeredRepository {
...
+        let init_logical_size =
+            if let Ok(src_pgdir) = tenant_mgr::get_local_timeline_with_load(self.tenant_id, src) {
+                let logical_size = src_pgdir.get_current_logical_size();
+                // Check LSN after gettig logical size to exclude rac condition:
```
Comment (aome510) [inline pageserver/src/layered_repository.rs:331]:
```suggestion
                // Check LSN after getting logical size to exclude race condition
```

File: pageserver/src/layered_repository.rs [hunk 3]
```diff
@@ -1099,6 +1129,10 @@ pub struct LayeredTimeline {
     // It can be unified with latest_gc_cutoff_lsn under some "first_valid_lsn",
     // though lets keep them both for better error visibility.
     initdb_lsn: Lsn,
+
+    // Initial logical size of timeline (if known).
+    // Logical size can be copied from ancestor timelien when new branch is create at last LSN
```
Comment (aome510) [inline pageserver/src/layered_repository.rs:1134]:
```suggestion
    // Logical size can be copied from ancestor timeline when new branch is create at last LSN
```

**Review by aome510 — APPROVED** [review]
LGTM beside some minor typos

**Pull request merged**